codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/CodegenVisitor.java

```diff
@@ -255,6 +255,9 @@ void execute() {
         List<SymbolDependency> dependencies = writers.getDependencies();
         writers.flushWriters();
 
+        LOGGER.fine("Generating go.mod file");
```

Comment:
This log should be moved to `GoModGenerator.java` after the `generateGoMod` check.